File: src/main/java/org/jukeboxmc/block/LiquidFlowEvent.java
package org.jukeboxmc.block;

import org.jukeboxmc.event.CancelableBlockEvent;

/**
 * @author LucGamesYT
 * @version 1.0
 */
public class LiquidFlowEvent extends CancelableBlockEvent {

    private Block blockTo;
    private BlockLiquid source;
    private int newFlowDecay;

    public LiquidFlowEvent( Block blockTo, BlockLiquid source, int newFlowDecay ) {
        super( blockTo );
        this.blockTo = blockTo;
        this.source = source;
        this.newFlowDecay = newFlowDecay;
    }

    public Block getBlockTo() {
        return this.blockTo;
    }

    public BlockLiquid getSource() {
        return this.source;
    }

    public int getNewFlowDecay() {
        return this.newFlowDecay;
    }
}

File: JukeboxMC-API/src/main/kotlin/org/jukeboxmc/api/event/block/LiquidFlowEvent.kt
package org.jukeboxmc.api.event.block

import org.jukeboxmc.api.block.Block
import org.jukeboxmc.api.block.Liquid
import org.jukeboxmc.api.event.Cancellable

class LiquidFlowEvent(
    block: Block,
    private val source: Liquid,
    private val newFlowDecay: Int
) : BlockEvent(block), Cancellable {

    fun getSource(): Liquid {
        return this.source
    }

    fun getNewFlowDecay(): Int {
        return this.newFlowDecay
    }


}